Audio connection pool

jsarabia opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bible-translation-tools/orature/343)
<!-- Reviewable:end -->

jsarabia committed:
wip

jsarabia committed:
wip

jsarabia committed:
Update AudioConnectionFactory.kt

jsarabia committed:
Keep progress working

Audio Controller needs more work; disposable is now not cleaned up, but cleaning it up here was a bug that wouldn't reattach

jsarabia committed:
wip

jsarabia committed:
Update AudioConnectionFactory.kt

jsarabia committed:
Merge branch 'dev' into jbs-audio-connection-pool

jsarabia committed:
Commit code from Max for device switching (#344)

* Commit code from Max for device switching

Code from Max's Device Switching PR, modified slightly to fit better with the audio connection pool.

* Extract AudioFormat to DI Builder

* Update Messages_en.properties

* Change audio output icon

* Bugfix: playback of marked sections

playback within restricted sections of a file (such as verses in a chapter file) do not need to be adjusted- this is taken care of within the reader. As a result, attempts at offsetting values due to starting position are necessary. For example, playing verse 2 when loaded from a chapter file will begin in the audio player at 0, not the frame position of the chapter file.

* Bugfix: glitch on pause/play button

isPlaying is tied to listeners, and actively setting them on button presses causes some conflict. This removes the explicit setting on playback state and instead defers completely to listeners.

* Code Organization

* Pass AudioConnectionFactory to Plugins (#346)

* Pass AudioConnectionFactory to Plugins

* Provide Audio Recorder over Connection Factory (#347)

jsarabia committed:
Merge branch 'dev' into jbs-audio-connection-pool

jsarabia committed:
Merge conflict issues

jsarabia committed:
Fix audio slider drag bug

Fixes a bug where dragging would prevent the slider from updating again

jsarabia committed:
Update Messages_en.properties

jsarabia committed:
Bugfix: two audio recorder implementations

Replaces the AudioRecorder in the device package with the one from the recorder app. Removes it from the recorder app. The device package audio recorder seems to be incomplete/outdated/unused.

jsarabia committed:
Fix plugin test

Needed to mock the connection factory

jsarabia committed:
Remove unused imports

mXaln reviewed (changes requested):
> Reviewed 28 of 28 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/343)* status: all files reviewed, 2 unresolved discussions (waiting on @jsarabia)

___
*[jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/controllers/AudioPlayerController.kt, line 60 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/343#-MjF6qZ0-0-lmp3wp5oZ:-MjF6qZ0-0-lmp3wp5o_:b-2r0el1) ([raw file](https://github.com/bible-translation-tools/orature/blob/3c2833bd5c4f270533638dbf7af1d57951119cf9/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/controllers/AudioPlayerController.kt#L60)):*
> ```Kotlin
>                 disposable = startProgressUpdate()
>                 play()
>                 _player.addEventListener {
> ```

Adding new listener every time we toggle playing concerns me a bit. I would add listener once we have player object, it means in load() method
___
*[jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceAudioFragment.kt, line 33 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/343#-MjTyOTC7RJ55yF1oHle:-MjTyOTC7RJ55yF1oHlf:btas2z7) ([raw file](https://github.com/bible-translation-tools/orature/blob/3c2833bd5c4f270533638dbf7af1d57951119cf9/jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceAudioFragment.kt#L33)):*
> ```Kotlin
> 
> //    @Inject
> //    lateinit var player: IAudioPlayer
> ```

Do we need this commented out code in this class?


<!-- Sent from Reviewable.io -->

jsarabia committed:
Try mocking dependency graph for tests

jsarabia committed:
Comment out SettingsViewModelTest for now

Can't get the SettingsViewModelTest to work when audio devices are not on the device running tests. Need to configure the AudioModule components at run time instead of in the companion object for that to happen.

jsarabia committed:
Remove Source Audio from Marker App

Source Audio isn't used in the Marker App, only the source text.

jsarabia committed:
Set Progress listener on loading audio player, not on Play

jsarabia reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/343)* status: 25 of 31 files reviewed, 2 unresolved discussions (waiting on @mXaln)

___
*[jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/controllers/AudioPlayerController.kt, line 60 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/343#-MjF6qZ0-0-lmp3wp5oZ:-MjUmpQd1tN6fDyb5q0e:b-21w2q2) ([raw file](https://github.com/bible-translation-tools/orature/blob/3c2833bd5c4f270533638dbf7af1d57951119cf9/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/controllers/AudioPlayerController.kt#L60)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

Adding new listener every time we toggle playing concerns me a bit. I would add listener once we have player object, it means in load() method
</blockquote></details>

good idea, done!
___
*[jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceAudioFragment.kt, line 33 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/343#-MjTyOTC7RJ55yF1oHle:-MjUkhUoAiejwszrkaWv:b-yr2s75) ([raw file](https://github.com/bible-translation-tools/orature/blob/3c2833bd5c4f270533638dbf7af1d57951119cf9/jvm/markerapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/markerapp/app/view/SourceAudioFragment.kt#L33)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

Do we need this commented out code in this class?
</blockquote></details>

Oh whoops! You're right; a lot of code can be pulled from here since the Marker app only uses source text. I've Changed the name as well.


<!-- Sent from Reviewable.io -->

jsarabia committed:
Update TestPersistenceComponent.kt

jsarabia committed:
Update AudioPlayerController.kt

mXaln reviewed (approved):
> Reviewed 3 of 3 files at r3, 2 of 2 files at r4, 1 of 1 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/343)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jsarabia)



<!-- Sent from Reviewable.io -->

jsarabia committed:
Mock AudioModule for integration test
